elasticai/creator/vhdl/skeleton_id.py

```diff
@@ -0,0 +1,146 @@
+"""
+The skeleton id should be computed and set after all other code is generated.
+"""
+
+import logging
+from collections.abc import Iterable
+from hashlib import blake2b, file_digest
+from importlib.metadata import version as _pypackage_version
+from itertools import filterfalse
+from pathlib import Path
+
+import tomlkit as toml
+from tomlkit.toml_file import TOMLFile
+
+
+def _is_id(line: str):
+    return line.strip().startswith("constant SKELETON_ID")
+
+
+def _build_skeleton_id_line(id: Iterable[str]) -> str:
+    id_str = ", ".join(f'x"{b.upper()}"' for b in id)
+
+    return f"  constant SKELETON_ID : skeleton_id_t := ({id_str});"
+
+
+def replace_id_in_code(code: Iterable[str], id: Iterable[str]) -> Iterable[str]:
+    """
+    Look for a line that starts with `constant SKELETON_ID` and replace it with
+    the given id.
+    """
+    for line in code:
+        if _is_id(line):
+            yield _build_skeleton_id_line(id)
+            yield "\n"
+        else:
+            yield line
+
+
+class SkeletonIdHash:
+    SIZE: int = 16
+
+    def __init__(self):
+        self._digests = []
```

Comment:
What is the reason to not move that out of init to SIZE (line 40)

Reply:
Because it's only needed inside this class. Everyone else can just call `len(digest)` if they want to know. Should probably call that `_SIZE: int = 16`